setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php

```diff
@@ -43,7 +49,14 @@ public function addExcludePatterns(array $excludePatterns)
      */
     public function getList($path)
     {
+
         $realPath = realpath($path);
+        $isGeneration = strpos($realPath, DIRECTORY_SEPARATOR . 'generation' . DIRECTORY_SEPARATOR) !== false;
```

Comment:
I'm working on refactoring this part of the code, there is some interaction with some of the tests somehow that did not like this.  I had intended to use DirectoryList class to get the full path but introducing that dependency broke lots of other tests so I opted to just do a string match.  I will make the check more intentional.